TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/automation/TelepresenceBotAccessibilityService.java

```diff
@@ -0,0 +1,54 @@
+package com.novoda.tpbot.automation;
+
+import android.accessibilityservice.AccessibilityService;
+import android.util.Log;
+import android.view.accessibility.AccessibilityEvent;
+import android.view.accessibility.AccessibilityNodeInfo;
+
+import java.util.List;
+
+public class TelepresenceBotAccessibilityService extends AccessibilityService {
+
+    private static final String HANGOUTS_PACKAGE_NAME = "com.google.android.talk";
+    private static final String JOIN_VIDEO_CALL = "JOIN VIDEO CALL";
+
+    @Override
+    public void onAccessibilityEvent(AccessibilityEvent event) {
+        AccessibilityNodeInfo source = event.getSource();
+
+        if (source == null) {
+            return;
+        }
+
+        String packageName = String.valueOf(source.getPackageName());
+        int eventType = event.getEventType();
+
+        if (isHangouts(packageName) && isWindowStateChangeEvent(eventType)) {
```

Comment:
`WindowStateChange` is one of the first events to trigger when navigating from `tpBot` to the `Hangouts` application, it contains the full view hierarchy via `AccessibilityNodes`.